src/code/z_skin.c

```diff
@@ -1,23 +1,272 @@
+        Skin_ApplyLimbModifications(gfxCtx, skin, limbIndex, arg3);
+    }
+
+    gSPDisplayList(POLY_OPA_DISP++, data->dlist);
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+/**
+ * Draw a limb of type SKIN_LIMB_TYPE_NORMAL, of the skeleton `skin` at index `limbIndex`
+ */
+void Skin_DrawLimb(GraphicsContext* gfxCtx, Skin* skin, s32 limbIndex, Gfx* dlistOverride, s32 drawFlags) {
+    Gfx* gfx = dlistOverride;
+    SkinLimb** skeleton;
+
+    OPEN_DISPS(gfxCtx);
+
+    skeleton = Lib_SegmentedToVirtual(skin->skeletonHeader->segment);
+
+    if (dlistOverride == NULL) {
+        gfx = ((SkinLimb*)Lib_SegmentedToVirtual(skeleton[limbIndex]))->segment;
+    }
+
+    if (gfx != NULL) {
+        Mtx* mtx = SkinMatrix_MtxFToNewMtx(gfxCtx, &gSkinLimbMatrices[limbIndex]);
+
+        if (mtx != NULL) {
+            gSPMatrix(POLY_OPA_DISP++, mtx, G_MTX_PUSH | G_MTX_MUL | G_MTX_MODELVIEW);
+            gSPDisplayList(POLY_OPA_DISP++, gfx);
+            gSPPopMatrix(POLY_OPA_DISP++, G_MTX_MODELVIEW);
+            gDPPipeSync(POLY_OPA_DISP++);
+        }
+    }
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+void Skin_DrawImpl(Actor* actor, GlobalContext* globalCtx, Skin* skin, SkinPostDraw postDraw,
+                   SkinOverrideLimbDraw overrideLimbDraw, s32 setTranslation, s32 arg6, s32 drawFlags) {
+    s32 i;
+    SkinLimb** skeleton;
+    GraphicsContext* gfxCtx = globalCtx->state.gfxCtx;
+
+    OPEN_DISPS(gfxCtx);
+
+    if (!(drawFlags & SKIN_DRAW_FLAG_CUSTOM_TRANSFORMS)) {
+        Skin_ApplyAnimTransformations(skin, gSkinLimbMatrices, actor, setTranslation);
+    }
+
+    skeleton = Lib_SegmentedToVirtual(skin->skeletonHeader->segment);
+
+    if (!(drawFlags & SKIN_DRAW_FLAG_CUSTOM_MATRIX)) {
+        Mtx* mtx;
+
+        gSPMatrix(POLY_OPA_DISP++, &gIdentityMtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        mtx = SkinMatrix_MtxFToNewMtx(gfxCtx, &skin->mtx);
+        if (mtx == NULL) {
+            goto close_disps;
+        }
+        gSPMatrix(POLY_OPA_DISP++, mtx, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    }
+
+    for (i = 0; i < skin->skeletonHeader->limbCount; i++) {
+        s32 shouldDraw = true;
+        s32 segmentType;
+
+        if (overrideLimbDraw != NULL) {
+            shouldDraw = overrideLimbDraw(actor, globalCtx, i, skin);
+        }
+
+        segmentType = ((SkinLimb*)Lib_SegmentedToVirtual(skeleton[i]))->segmentType;
+
+        if (segmentType == SKIN_LIMB_TYPE_ANIMATED && shouldDraw) {
+            Skin_DrawAnimatedLimb(gfxCtx, skin, i, arg6, drawFlags);
+        } else if (segmentType == SKIN_LIMB_TYPE_NORMAL && shouldDraw) {
+            Skin_DrawLimb(gfxCtx, skin, i, NULL, drawFlags);
+        }
+    }
+
+    if (postDraw != NULL) {
+        postDraw(actor, globalCtx, skin);
+    }
+
+close_disps:;
+    CLOSE_DISPS(gfxCtx);
+}
+
+// allows specifying postDraw and setTranslation
+void func_80138228(Actor* actor, GlobalContext* globalCtx, Skin* skin, SkinPostDraw postDraw, s32 setTranslation) {
```

Comment:
```suggestion
/**
 * Allows specifying postDraw and setTranslation
 */
void func_80138228(Actor* actor, GlobalContext* globalCtx, Skin* skin, SkinPostDraw postDraw, s32 setTranslation) {
```
I think this is more the style for explanations of what a function does? (applies to the functions below as well)

Reply:
at least in oot these are meant to be temporary and aid with actually naming the functions